tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodAttributeExecuteIntegration.cs

```diff
@@ -87,49 +87,75 @@ internal static CallTargetState OnMethodBegin<TTarget, TTestMethod>(TTarget inst
             var duration = testMethodState.Elapsed;
             var testMethod = testMethodState.TestMethod;
             var isEfdTest = false;
+            var isAttemptToFix = false;
             var allowRetries = false;
             var resultStatus = TestStatus.Skip;
-            if (returnValue is IList { Count: > 0 } returnValueList)
+
+            if (!(returnValue is IList { Count: > 0 } returnValueList))
+            {
+                Common.Log.Warning("TestMethodAttributeExecuteIntegration: Failed to extract TestResult from return value");
+                testMethodState.Test.Close(TestStatus.Fail);
+                return new CallTargetReturn<TReturn?>(returnValue);
+            }
+
+            MsTestIntegration.AddTotalTestCases(returnValueList.Count - 1);
+            for (var i = 0; i < returnValueList.Count; i++)
             {
-                MsTestIntegration.AddTotalTestCases(returnValueList.Count - 1);
-                for (var i = 0; i < returnValueList.Count; i++)
+                var test = i == 0 ? testMethodState.Test : MsTestIntegration.OnMethodBegin(testMethodState.TestMethod, testMethodState.TestMethod.Type, isRetry: false, testMethodState.Test.StartTime);
+                if (test?.GetTags() is { } testTags)
                 {
-                    var test = i == 0 ? testMethodState.Test : MsTestIntegration.OnMethodBegin(testMethodState.TestMethod, testMethodState.TestMethod.Type, isRetry: false, testMethodState.Test.StartTime);
-                    if (test is not null)
+                    if (testOptimization.EarlyFlakeDetectionFeature?.Enabled == true)
                     {
-                        if (test.GetTags() is { } testTags && testOptimization.EarlyFlakeDetectionFeature?.Enabled == true)
+                        isEfdTest = isEfdTest || testTags.TestIsNew == "true";
+                        if (isEfdTest && duration.TotalMinutes >= 5)
                         {
-                            isEfdTest = isEfdTest || testTags.TestIsNew == "true";
-                            if (isEfdTest && duration.TotalMinutes >= 5)
-                            {
-                                testTags.EarlyFlakeDetectionTestAbortReason = "slow";
-                            }
+                            testTags.EarlyFlakeDetectionTestAbortReason = "slow";
                         }
+                    }
+
+                    if (testOptimization.TestManagementFeature?.Enabled == true)
+                    {
+                        isAttemptToFix = isAttemptToFix || testTags.IsAttemptToFix == "true";
+                    }
 
-                        resultStatus = HandleTestResult(test, testMethod, returnValueList[i], exception);
```

Comment:
previously, this was set regardless of the calls to flaky detection and/or `GetTags()`, but now it's skipped - is that correct? 🤔